src/core/api.c

```diff
@@ -1278,31 +1278,11 @@ MsQuicDatagramSend(
 
     QUIC_TEL_ASSERT(!Connection->State.Freed);
 
-    if (!Connection->Datagram.SendEnabled) {
-        QuicTraceEvent(
-            ConnError,
-            "[conn][%p] ERROR, %s.",
-            Connection,
-            "Datagrams not enabled by peer");
-        Status = QUIC_STATUS_NOT_SUPPORTED;
-        goto Error;
```

Comment:
Isn't it better to fail early so the app knows not to try to send datagrams?

Reply:
It's a matter of synchronization. Also, it's not worth optimizing for the negative case here, IMO. If they're incorrectly trying to queue datagrams, I'm Ok if it ends up doing a pool alloc followed by a pool free.

Reply:
How is the failure bubbled up to the app, if at all?

Reply:
`QuicDatagramQueueSend` will fail the error back to the app.